Formula/gitlab-gem.rb

```diff
@@ -0,0 +1,30 @@
+class GitlabGem < Formula
+  desc "Ruby client and CLI for GitLab API"
+  homepage "https://github.com/NARKOZ/gitlab"
+  url "https://github.com/NARKOZ/gitlab/archive/v4.3.0.tar.gz"
+  sha256 "b2679d0696642f67ba84a4f9df21ad4a7bf1bc814d7f86218b72ebe5e56ff2d0"
+  head "https://github.com/NARKOZ/gitlab.git"
+
+  resource "httparty" do
+    url "https://rubygems.org/gems/httparty-0.16.0.gem"
+    sha256 "c3e08fac9079fdbe175158782c61f6db6a1918446399560d0bfce1e752b5a5d2"
+  end
+
+  resource "terminal-table" do
+    url "https://rubygems.org/gems/terminal-table-1.8.0.gem"
+    sha256 "13371f069af18e9baa4e44d404a4ada9301899ce0530c237ac1a96c19f652294"
+  end
+
+  def install
+    ENV["GEM_HOME"] = libexec
+    system "gem", "build", "gitlab.gemspec"
+    system "gem", "install", "gitlab-#{version}.gem"
+    bin.install "exe/gitlab"
+    bin.env_script_all_files(libexec/"exe", :GEM_HOME => ENV["GEM_HOME"])
+    libexec.install Dir["*"]
+  end
+
+  test do
+    assert_equal "Gitlab Ruby Gem 4.3.0", shell_output("#{bin}/gitlab --version").strip
```

Comment:
Could you provide an example of better test?

Reply:
What is a formula supposed to be testing? I am assuming that it is only testing if install was successful and not if the utility works. In other words, since the application that the formula is installing should handle its own tests, shouldn't this just be testing installation? If that is the case, then what is wrong with this test? That string comes from the application itself and wouldn't output if the application installation was unsuccessful.

Reply:
We want to have a test, not only that the application is installed, but that it is functional (to some level). Tests of `-h` or `--version` tend to miss some possible issues, and lack detecting brokenness when we test new versions.

A test could perform an operation, either locally, or try to perform a remote operation (querying the state of something, or checking out a known project, etc.). Or, if this is not possible, and the application can only work with authentication, check for expected failure due to lack of credentials.